File: src/ios/Sources/AnsightCore/PairingEnrollment.swift
import Foundation

public struct PairingEnrollment: Sendable, Codable, Equatable {
    public var ticketId: String
    public var secret: String
    public var expiresAt: String
    public var grantExpiresAt: String
    public var maxUses: Int
    public var maxScopes: [String]
    public var allowCritical: Bool

    public init(
        ticketId: String,
        secret: String,
        expiresAt: String,
        grantExpiresAt: String,
        maxUses: Int = 1,
        maxScopes: [String] = ["Read"],
        allowCritical: Bool = false
    ) {
        self.ticketId = ticketId
        self.secret = secret
        self.expiresAt = expiresAt
        self.grantExpiresAt = grantExpiresAt
        self.maxUses = maxUses
        self.maxScopes = maxScopes
        self.allowCritical = allowCritical
    }
}

File: src/dotnet/Ansight.Core/Pairing/Models/PairingEnrollment.cs
using System.Text.Json.Serialization;

namespace Ansight.Pairing.Models;

/// <summary>
/// Access token and registration lifetime carried by an enrollment QR.
/// </summary>
public sealed class PairingEnrollment
{
    [JsonPropertyName("accessToken")]
    public required string Secret { get; set; }

    public required DateTimeOffset ExpiresAt { get; set; }

    public required DateTimeOffset GrantExpiresAt { get; set; }

    public int MaxUses { get; set; } = 1;

    public string[] MaxScopes { get; set; } = ["Read"];

    public bool AllowCritical { get; set; }
}
